includes/services/Performer.php

```diff
@@ -159,15 +163,31 @@ public function run($objectName, $objectType, array $vars = []): string
         // execute main file with callbacks
         $files = array_merge($object['before_callbacks'], [$object], $object['after_callbacks']);
         foreach ($files as $file) {
-            if ($file['isDefinedAsClass']) {
-                $performable = $this->createPerformable($file, $vars, $output);
-                $output .= $performable->run();
-            } else {
-                $vars['plugin_output_new'] = &$output;
-                // need to run them from YesWiki Class so the variable $this (used in all the plain PHP object) refers to YesWiki, not to Performer service
-                $vars = $this->wiki->runFileInBuffer($file['filePath'], $vars);
-                $output = &$vars['plugin_output_new'];
-                unset($vars['plugin_output_new']);
+            try {
+                if ($file['isDefinedAsClass']) {
+                    $performable = $this->createPerformable($file, $vars, $output);
+                    $output .= $performable->run();
+                } else {
+                    $vars['plugin_output_new'] = &$output;
+                    // need to run them from YesWiki Class so the variable $this (used in all the plain PHP object) refers to YesWiki, not to Performer service
+                    $vars = $this->wiki->runFileInBuffer($file['filePath'], $vars);
+                    $output = &$vars['plugin_output_new'];
+                    unset($vars['plugin_output_new']);
+                }
+            } catch (Throwable $t) {
+                // catch all errors and exceptions thrown by the execution of the performable
+                $message = [
+                    'type' => 'danger',
+                    // display a generic message with the detailled error
+                    'message' => _t('PERFORMABLE_ERROR') . "<br/>" . $t->getMessage()
+                ];
+                if ($objectType == 'handler' || $objectType == 'formatter') {
```

Comment:
Après réflexion ce WE, je me demande si c'est pertinent de faire `renderInSquelette` pour les `formatter`.
En général, les `formatter` sont appelés par des actions ou des handlers, donc en faisant remonter les Header et Footer pour les erreurs de formatter on risque d'avoir deux fois le header.

Reply:
Yes t'as raison, je sais pas pourquoi je l'ai mis, au début je voulais mettre seulement 'handler' puis au dernier moment j'ai changé.
Les formatters étant appelés par la méthode $wiki->Format(...), ils sont effectivement affichés "en ligne".
J'ai fais le commit, thx